src/assignments/coded-messages.rst

```diff
@@ -0,0 +1,336 @@
+      :header-rows: 1
+
+      * - ``letter``
+        - Return Value
+      * - a or A
+        - 0
+      * - d or D
+        - 3
+      * - z or Z
+        - 25
+
+#. *Don't forget to test your function!* In the ``main()`` function, call
+   ``alphabet_position`` and send in different letters as arguments. Be sure to
+   print the returned result to check your code.
+
+.. admonition:: Tip
+
+   #. You can assume the argument sent to the function will be a single letter.
+      Do not worry about what to do with longer strings or non-letter
+      characters.
+   #. ``alphabet_position`` should NOT print anything. It needs to *return* a
+      value instead.
+
+The ``shift_character`` Function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Define the function ``shift_character(char, shift)`` with two parameters.
+   ``char`` accepts a *character* (a string of length 1) and ``shift`` accepts
+   an integer value.
+#. If ``char`` is a letter, the function returns a different one. The new
+   letter will be ``shift`` places to the right in the alphabet.
+#. The case of the original letter and the returned value should match. For
+   example, ``shift_character("A", 1)`` returns ``"B"``, instead of ``"b"``.
+#. If ``char`` is NOT a letter, then the function returns the same character.
+
+   .. list-table:: Sample Results
+      :header-rows: 1
+
+      * - ``char``
+        - ``shift``
+        - Return Value
+      * - a
+        - 13
+        - n
+      * - D
+        - 4
+        - H
+      * - z
+        - 3
+        - c
+      * - J
+        - 30
+        - N
+      * - ?, 6, &, etc.
+        - (any integer)
+        - ?, 6, &, etc. (no change for non-letter characters)
+
+#. *Test your function!* In ``main()``, call ``shift_character`` and send in
+   different characters and shift values. Be sure to print the letter returned
+   to make sure your code works properly before moving on.
+
+.. admonition:: Tip
+
+   #. You should make use of your ``alphabet_position`` function! Thinking in
+      terms of index values helps.
+   #. Notice that the function "wraps around" when it reaches the end of the
+      alphabet. If a shift moves the program past the letter ``'z'`` (index
+      value ``25``), it moves back to the start of the alphabet and continues
+      counting with ``'a'`` (index value ``0``).
+   #. The :ref:`modulus operator <modulo>` will be a BIG help here to keep the
+      index values in the range 0 - 25.
+
+The ``build_code_dict`` Function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Define the function ``build_code_dict`` that takes one parameter (an
+   integer).
+#. The function will build and return a dictionary. Each of the keys will be a
+   letter from the alphabet. Each of the values will also be a letter, but
+   shifted to the right a certain number of places. How many places? That
+   depends on the integer sent to the function.
+
+   *Hints*: The accumulator pattern strikes again! Also, notice how you can
```

Comment:
maybe replace 'notice' with 'think about' since there is no example to turn to

Reply:
remember that gives the impression that we've previously told the reader to use the shift_character function in this way. I think 'think about' fits better

Reply:
Done!